src/GLX/libglx.c

```diff
@@ -1275,13 +1275,13 @@ static char *MergeExtensionStrings(char *currentString, const char *newString)
     name = newString;
     nameLen = 0;
     while (FindNextExtensionName(&name, &nameLen)) {
-        if (!IsExtensionInString(currentString, name, nameLen)) {
+        if (!IsExtensionInString(buf, name, nameLen)) {
             *ptr++ = ' ';
             memcpy(ptr, name, nameLen);
             ptr += nameLen;
+            *ptr = '\0';
```

Comment:
Why move this line? The next pass through the while loop would overwrite the null terminator, so I think it's equivalent to just adding the null terminator after the loop.